pytorch_lightning/trainer/trainer.py

```diff
@@ -1042,44 +1039,45 @@ def test(
 
     def check_model_configuration(self, model: LightningModule):
         r"""
-        Checks that the model is configured correctly before training is started.
+        Checks that the model is configured correctly before training or testing is started.
 
         Args:
-            model: The model to test.
+            model: The model to check the configuration.
 
         """
         # Check training_step, train_dataloader, configure_optimizer methods
-        if not self.is_overridden('training_step', model):
-            raise MisconfigurationException(
-                'No `training_step()` method defined. Lightning `Trainer` expects as minimum a'
-                ' `training_step()`, `train_dataloader()` and `configure_optimizers()` to be defined.')
-
-        if not self.is_overridden('train_dataloader', model):
-            raise MisconfigurationException(
-                'No `train_dataloader()` method defined. Lightning `Trainer` expects as minimum a'
-                ' `training_step()`, `train_dataloader()` and `configure_optimizers()` to be defined.')
-
-        if not self.is_overridden('configure_optimizers', model):
-            raise MisconfigurationException(
-                'No `configure_optimizers()` method defined. Lightning `Trainer` expects as minimum a'
-                ' `training_step()`, `train_dataloader()` and `configure_optimizers()` to be defined.')
-
-        # Check val_dataloader, validation_step and validation_epoch_end
-        if self.is_overridden('val_dataloader', model):
-            if not self.is_overridden('validation_step', model):
-                raise MisconfigurationException('You have passed in a `val_dataloader()`'
-                                                ' but have not defined `validation_step()`.')
+        if not self.testing:
+            if not self.is_overridden('training_step', model):
```

Comment:
I would consider moving this to `__check_model_configuration_test` and alternatively bellow

Reply:
I suggest removing `check_testing_model_configuration` because it's just a repeated code and testing configuration can be handled within `check_model_configuration`. Also during testing `check_model_configuration` is called which is unnecessary and was not working in #1720. The PR fixes both the issues.